runtime/array.c

```diff
@@ -383,8 +340,7 @@ static value caml_array_gather(intnat num_arrays,
   int isfloat = 0;
```

Comment:
Is this optimisation safe according to the memory model? See comment in https://github.com/ocaml/ocaml/pull/10831/files#r776927681.

Reply:
Fixed by https://github.com/ocaml-multicore/ocaml-multicore/pull/822.